web/src/notifications.js

```diff
@@ -241,7 +242,12 @@ export function process_notification(notification) {
     const $content = $("<div>").html(message.content);
     ui.replace_emoji_with_text($content);
     spoilers.hide_spoilers_in_notification($content);
-    content = $content.text();
+
+    if ($content.text().trim() === "" && message_has_attachment) {
```

Comment:
Did you verify that `message_has_attachment` works? It will not due to obvious reasons.

Reply:
@amanagr I was able to generate a notification with content `(attached file)` with this. Is there any other way I should verify this?

Reply:
I didn't give it a lot of thought because it was working but I know what is the mistake, let me fix it.

Reply:
That is why I need you to figure out how to write a test case for this which includes cases which are valid messages but `$content.text().trim() === ""` like `.